### ISSUE TITLE
Design module architecture

### ISSUE THREAD

**Issue by isurunuwanthilaka** [isurunuwanthilaka/log4b#1]
I have done a research on what APIs available for writing this ballerina logger. But found out no APIs to access stderr. If we are going to write ballerina itself, only IO APIs available. So only messages passed to logger can be logged. Other exceptions cannot be caught as no way to access stderr stream.

What do you thinks ? @gayannagc

**Comment by isurunuwanthilaka** [comment]
Already ballerina team is migrating code to JDK 11, so java will be with ballerina for quite some time, I suppose.It will be okay to write this module with Java 11 and ballerina. Therefore lets look into ballerina/log and ballerina/java modules, then we can try implementation. @gayannagc

**Comment by gayannagc** [comment]
yeah if that's the case continuing with ballerina/java approach would be better.
